systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java

```diff
@@ -426,6 +430,30 @@ public static String getImageValueFromCO(String name) {
         return resource;
     }
 
+    public static <T extends HasMetadata> long getResourceOperationTimeout(String kind, ResourceOperation operation) {
+        return getResourceOperationTimeout(kind, operation, 0);
+    }
+
+    public static <T extends HasMetadata> long getResourceOperationTimeout(String kind, ResourceOperation operation, int podsToRoll) {
+        List<String> shortTimeout = asList(KafkaConnector.RESOURCE_KIND, KafkaTopic.RESOURCE_KIND, KafkaUser.RESOURCE_KIND);
+        long timeout = 0;
+
+        switch (operation) {
+            case RESOURCE_READY:
+            case RESOURCE_CREATION:
+                if (shortTimeout.contains(kind)) {
+                    timeout = Constants.TIMEOUT_FOR_TOPIC_CREATION;
+                } else {
+                    timeout = Constants.TIMEOUT_FOR_RESOURCE_CREATION;
+                }
+                break;
```

Comment:
I don't think this is a good approach. `Kafka` status set to `ready` could take much more time than simple CR creation. You should use different timeouts based on resource type probably.